ConsoleApp1/ConsoleApp1/Program.cs

```diff
@@ -0,0 +1,528 @@
+                    var arr = Apples.ToArray();
+                    Apples.Clear();
+                    for (int i = 0; i < arr.Length; i++)
+                    {
+                        Apples.Add(arr[i]);
+                    }
+                    return;
+                }
+                count++;
+            }
+        }
+        public static class Snake
+        {
+            public static List<Point> Points = new List<Point>();
+            public static Rotate Rotation = Rotate.Right;
+            public static void Init()
+            {
+                Points.Clear();
+                Points.Add(new Point(3, 2));
+                Points.Add(new Point(3, 3));
+            }
+
+            public static void Turn(Rotate r)
+            {
+                if (r == Rotate.Up && Rotation == Rotate.Down) return;
+                if (r == Rotate.Left && Rotation == Rotate.Right) return;
+                if (r == Rotate.Down && Rotation == Rotate.Up) return;
+                if (r == Rotate.Right && Rotation == Rotate.Left) return;
+                CreatePoint(r);
+            }
+
+            internal static void CreatePoint(Rotate r)
+            {
+                var ar = Points.ToArray();
+                int x = ar[ar.Length - 1].x;
+                int y = ar[ar.Length - 1].y;
+                Points = ar.ToList();
+                switch (r)
+                {
+                    case Rotate.Up:
+                        {
+                            x--;
+                        }
+                        break;
+                    case Rotate.Down:
+                        {
+                            x++;
+                        }
+                        break;
+                    case Rotate.Left:
+                        {
+                            y--;
+                        }
+                        break;
+                    case Rotate.Right:
+                        {
+                            y++;
+                        }
+                        break;
+                }
+                Points.Add(new Point(x, y));
+
+            }
+
+
+            public static void RemoveLast()
+            {
+                var ar = Points.ToArray();
+                int x = ar[0].x;
+                int y = ar[0].y;
+                Points = ar.ToList();
+                Points.RemoveAt(0);
+                Matrix.Screen[x, y] = ".";
+            }
+
+            public static bool Scan(Rotate r, out bool eat, out int ax, out int ay)
+            {
+                var ar = Points.ToArray();
+                int x = ar[ar.Length - 1].x;
+                int y = ar[ar.Length - 1].y;
+                ax = x;
+                ay = y;
+                Points = ar.ToList();
+                try
+                {
+                    switch (r)
+                    {
+                        case Rotate.Up:
+                            if (Matrix.Screen[x - 1, y] == "." || Matrix.Screen[x - 1, y] == "O")
```

Comment:
it's so hard to read, I think there is better option to do it (at least extract to method)